File: examples/java/documentation/src/main/java/com/mycompany/app/assets/MainAssets.java
package com.mycompany.app.assets;

// DOCS_BLOCK_START:assets
import software.constructs.Construct;
import com.hashicorp.cdktf.App;
import com.hashicorp.cdktf.TerraformStack;
import com.hashicorp.cdktf.TerraformAsset;
import com.hashicorp.cdktf.TerraformAssetConfig;
import com.hashicorp.cdktf.AssetType;
import imports.aws.AwsProvider;
import imports.aws.AwsProviderConfig;
import imports.aws.s3.S3Bucket;
import imports.aws.s3.S3BucketConfig;
import imports.aws.s3.S3BucketObject;
import imports.aws.s3.S3BucketObjectConfig;

import java.nio.file.Paths;

public class MainAssets extends TerraformStack {
    public MainAssets(Construct scope, String name){
        super(scope, name);

        new AwsProvider(this,"aws", AwsProviderConfig.builder()
                .region("eu-central-1")
                .build()
        );

        S3Bucket bucket = new S3Bucket(this, "bucket", S3BucketConfig.builder()
                .bucket("demo")
                .build()
        );

        TerraformAsset asset = new TerraformAsset(this, "lambda-asset", TerraformAssetConfig.builder()
                .path(Paths.get(System.getProperty("user.dir"), "lambda").toString())
                .type(AssetType.ARCHIVE)
                .build()
        );

        new S3BucketObject(this, "lambda-archive", S3BucketObjectConfig.builder()
                .bucket(bucket.getBucket())
                .key(asset.getFileName())
                .source(asset.getPath())
                .build()
        );

    }

    public static void main(String[] args) {
        final App app = new App();
        new MainAssets(app, "demo");
        app.synth();
    }
}
// DOCS_BLOCK_END:assets

File: examples/python/documentation/main-assets.py
# DOCS_BLOCK_START:assets
from constructs import Construct
import cdktf
import imports.aws as aws
import os


class MyAssetStack(cdktf.TerraformStack):
    def __init__(self, scope: Construct, name: str):
        super().__init__(scope, name)

        aws.provider.AwsProvider(self, 'aws', region='eu-central-1')

        bucket = aws.s3_bucket.S3Bucket(self, "bucket", bucket="demo")

        asset = cdktf.TerraformAsset(self, "lambda-asset",
                                     path=os.path.join(os.path.dirname(
                                         __file__), 'lambda'),
                                     type=cdktf.AssetType.ARCHIVE
                                     )

        aws.s3_bucket_object.S3BucketObject(self, "lambda-archive",
                                            bucket=bucket.bucket,
                                            key=asset.file_name,
                                            source=asset.path
                                            )


app = cdktf.App()
MyAssetStack(app, "demo")

app.synth()
# DOCS_BLOCK_END:assets
